map/cMap.cpp

```diff
@@ -1139,10 +1139,17 @@ bool cMap::isAtMapBoundaries(int cell) {
     return false;
 }
 
-void cMap::fixCoordinatesToBeWithinPlayableMap(int &x, int &y) {
-    FIX_BORDER_POS(x, y);
+void cMap::fixCoordinatesToBeWithinPlayableMap(int &x, int &y) const {
+    x = std::clamp(x, 1, getWidth() - 2);
+    y = std::clamp(y, 1, getHeight() - 2);
 }
 
+void cMap::fixCoordinatesToBeWithinMap(int &x, int&y) const {
```

Comment:
`std::clamp` is quite nice :)

Even better would be, at some point, to have it accept a `cPoint` and return a `cPoint` or something similar. For now we use the same mechanism, so thats fine.